substrate/primitives/api/proc-macro/src/runtime_metadata.rs

```diff
@@ -222,6 +227,24 @@ pub fn generate_impl_runtime_metadata(impls: &[ItemImpl]) -> Result<TokenStream2
 			.expect("Trait path should always contain at least one item; qed")
 			.ident;
 
+		let mut deprecations = vec![];
+
+		if impl_.attrs.iter().any(|x| x.path().is_ident("deprecated")) {
+			let deprecation = crate::utils::get_deprecation(&crate_, &impl_.attrs)?;
+			deprecations.push(quote! {(stringify!(#trait_name_ident), #deprecation)});
+		};
+
+		for method in &impl_.items {
+			if let ImplItem::Fn(method) = method {
+				if method.attrs.iter().any(|x| x.path().is_ident("deprecated")) {
+					let name = &method.sig.ident;
+					dbg!(name);
+					let deprecation = crate::utils::get_deprecation(&crate_, &method.attrs)?;
+					deprecations.push(quote! {(stringify!(#name), #deprecation)});
```

Comment:
I forgot, method names are unique or they are prefixed with the trait name in the runtime api?
If they are not unique and prefixed by the trait name we should do same here. Otherwise if 2 runtime API have a method with the same name then deprecation information will propagate to both.

You could even strongly type it to differentiate between trait deprecation and method deprecation, but it can be ok as it is.